src/ncp/posix/netif.cpp

```diff
@@ -114,6 +114,35 @@ void Netif::UpdateIp6UnicastAddresses(const std::vector<Ip6AddressInfo> &aAddrIn
     mIp6UnicastAddresses.assign(aAddrInfos.begin(), aAddrInfos.end());
 }
 
+void Netif::SetNetifState(bool aState)
+{
+    otbrError    error = OTBR_ERROR_NONE;
+    struct ifreq ifr;
+    bool         ifState = false;
+
+    VerifyOrExit(mIpFd >= 0);
+    memset(&ifr, 0, sizeof(ifr));
+    strncpy(ifr.ifr_name, mNetifName.c_str(), IFNAMSIZ - 1);
+    VerifyOrExit(ioctl(mIpFd, SIOCGIFFLAGS, &ifr) == 0, error = OTBR_ERROR_ERRNO);
+
+    ifState = ((ifr.ifr_flags & IFF_UP) == IFF_UP) ? true : false;
+
+    otbrLogInfo("[netif] Changing interface state to %s%s.", aState ? "up" : "down",
```

Comment:
Is the `[netif]` prefix common in this project? We should standardize on an approach.

Reply:
Removed `[netif]` since we already have the log tag `NETIF` in otbr. (The log prefix is copied from OT posix netif).